src/coreclr/src/jit/codegenarmarch.cpp

```diff
@@ -1367,39 +1367,33 @@ void CodeGen::genPutArgSplit(GenTreePutArgSplit* treeNode)
 //    The child of store is a multi-reg call node.
 //    genProduceReg() on treeNode is made by caller of this routine.
 //
-void CodeGen::genMultiRegCallStoreToLocal(GenTree* treeNode)
+void CodeGen::genMultiRegStoreToLocal(GenTree* treeNode)
 {
     assert(treeNode->OperGet() == GT_STORE_LCL_VAR);
-
-#if defined(TARGET_ARM)
-    // Longs are returned in two return registers on Arm32.
-    // Structs are returned in four registers on ARM32 and HFAs.
-    assert(varTypeIsLong(treeNode) || varTypeIsStruct(treeNode));
-#elif defined(TARGET_ARM64)
-    // Structs of size >=9 and <=16 are returned in two return registers on ARM64 and HFAs.
-    assert(varTypeIsStruct(treeNode));
-#endif // TARGET*
+    assert(varTypeIsStruct(treeNode) || varTypeIsMultiReg(treeNode));
+    GenTree* op1       = treeNode->gtGetOp1();
+    GenTree* actualOp1 = op1->gtSkipReloadOrCopy();
+    assert(op1->IsMultiRegNode());
+    unsigned regCount = op1->GetMultiRegCount();
 
     // Assumption: current implementation requires that a multi-reg
     // var in 'var = call' is flagged as lvIsMultiRegRet to prevent it from
     // being promoted.
     unsigned   lclNum = treeNode->AsLclVarCommon()->GetLclNum();
-    LclVarDsc* varDsc = &(compiler->lvaTable[lclNum]);
-    noway_assert(varDsc->lvIsMultiRegRet);
-
-    GenTree*     op1       = treeNode->gtGetOp1();
-    GenTree*     actualOp1 = op1->gtSkipReloadOrCopy();
-    GenTreeCall* call      = actualOp1->AsCall();
-    assert(call->HasMultiRegRetVal());
+    LclVarDsc* varDsc = compiler->lvaGetDesc(lclNum);
+    if (op1->OperIs(GT_CALL))
```

Comment:
It can be another multi-reg local.

Reply:
I see only calls under `if (data->gtSkipReloadOrCopy()->IsMultiRegCall())`, so is it for your future changes?
The header should be updated `//    The child of the store is a multi-reg call or local?`

Reply:
I'm going to update it simply to "The child of the store is a multi-reg node." since I hope that soon we will be supporting multi-reg returns from HW intrinsics.